### ISSUE TITLE
Virtual host issue

### ISSUE THREAD

**Issue by bigfoot90** [lgallard/qBittorrent-Client#29]
I'm using nginx configured as reverse proxy from domain to port on localhost.

The problem is that I'm unable to connect throught this client breacause it do not send the domain to the server.

Here are the requests log from your client to nginx:
```
11.22.33.44 - - [27/Apr/2015:22:30:14 +0200] "GET json/torrents HTTP/1.1" 400 166 "-" "-"
```

Here are the correct requests log from my browser using the web interface:
```
11.22.33.44 - - [27/Apr/2015:22:31:57 +0200] "GET /json/torrents?i90cabsg HTTP/1.1" 200 2080 "http://mydomain.com/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.90 Safari/537.36"
```

**Comment by lgallard** [comment]
Hi bigfoot90,

Can you send me your nginx settings to reproduce the reverse proxy and make my tests?

**Comment by bigfoot90** [comment]
Hi, thank you for responce, here is my config:
```
server {
    listen 80;
    server_name mydomain.com;

    access_log  /var/log/nginx/torrent_access.log;
    error_log  /var/log/nginx/torrent_error.log;

    location ~ / {
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_pass http://192.168.1.10:8080;
    }
}
```
I've replaced the real domain for obvious reasons.
You need to add ```mydomain.com``` to your ```/etc/hosts``` file.

**Comment by lgallard** [comment]
When trying to connect to nginx using your setting I got this error:

>'qbittorrentclientpro W/System.err﹕ Caused by: org.apache.http.ProtocolException: The server failed to respond with a valid HTTP response'

Do you have any idea of what nginx expects? Because I followed the qBittorrent WebUI API specification to connect to the server.

**Comment by bigfoot90** [comment]
hi, can you post your nginx logs?
on linux systems they are in ```/var/log/nginx/error.log```

also make sure the qBittorrent is running with web interface enabled,
and doucle check your **address ip** and **port** on ```proxy_pass``` parameter

**Comment by lgallard** [comment]
Hi bigfoot90,

I will be publishing a fix for this issue soon, after testing all the commands (add, delete, send torrents, etc).

**Comment by bigfoot90** [comment]
Thank you very much

**Comment by lgallard** [comment]
Hi bigfoot90,

You are welcome...it was fixed and published in version 3.9.2

Regards!

**Comment by bigfoot90** [comment]
Fantastic. It also works with HTTPS enabled on reverse proxy. Thanks.
